compiler/dspash/worker.py

```diff
@@ -41,15 +42,20 @@ def parse_exec_request(request):
     return request['cmd']
 
 def parse_exec_graph(request):
-    return request['graph'], request['shell_variables']
+    return request['graph'], request['shell_variables'], request['functions']
 
-def exec_graph(graph, shell_vars):
+def exec_graph(graph, shell_vars, functions):
     config.config['shell_variables'] = shell_vars
     script_path = to_shell_file(graph, config.pash_args)
 
     e = os.environ.copy()
     e['PASH_TOP'] = PASH_TOP
-    rc = subprocess.Popen(['bash', script_path], env=e)
+
+    # store functions
+    functions_file = create_filename(dir=config.PASH_TMP_PREFIX, prefix='pashFuncs')
+    write_file(functions_file, functions)
+    cmd = f"source {functions_file}; source {script_path}"
```

Comment:
Is this the subworker execution script (that executes a part of the graph)? Is there anyway we need to export all functions too?

Reply:
I don't think we need to re-export because declare -f outputs `declare -fx` statements for exported functions. I tried it and it seems to have the same effect as exporting the functions.